### PR TITLE
[develop]: Update python docstrings and generate preliminary technical documentation

### LINKED ISSUE #340: Text files in doc/ are outdated and redundant
## Description
[doc/INSTALL](https://github.com/ufs-community/ufs-srweather-app/blob/develop/doc/INSTALL) and [doc/RUNTIME](https://github.com/ufs-community/ufs-srweather-app/blob/develop/doc/RUNTIME) contain incomplete/outdated instructions for building and running the app. 

## Solution
Since they are redundant to other, more up-to-date instructions in the Users Guide, these files should be removed.

## Requirements
None

## Acceptance Criteria (Definition of Done)
Removal of text files `docs/INSTALL` and `docs/RUNTIME`

## Alternative Solutions (optional)
These files could be expanded and updated. This would be unhelpful in my opinion, as they are redundant and unnecessary.

### LINKED ISSUE #1130: Add Technical Documentation to SRW App docs
## Description
Discussion #1116 requested that Automated Technical Documentation be added to the SRW App. 
Previously, a [plan](https://docs.google.com/document/d/1c7m52Or6-XS-wXskJNmVOfG7Vt6i3ymo641sTcwNnZ4/edit#heading=h.msh2mblf6ie6) for automated technical documentation was developed. 

## Solution
As a first step, the docstrings for Python scripts in `ush` and `tests/WE2E` need to be updated, and documentation files for these scripts need to be added. 

## Requirements**
SRW App has API-like documentation of the `ush` and `tests/WE2E` scripts that is drawn from docstrings in those files.

## Acceptance Criteria (Definition of Done)
See "Requirements" section.

### PR DESCRIPTION
**Pull request by gspetro-NOAA**
## DESCRIPTION OF CHANGES: 
Initial implementation of SRW App technical documentation. Adds a section called "Technical Documentation" to the SRW App docs. 

This PR also removes outdated/redundant text files in `doc` (i.e., RUNTIME and INSTALL). 

### Type of change
- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

## TESTS CONDUCTED: 
Fundamental tests pass on Hera.

```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used
----------------------------------------------------------------------------------------------------
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta_2  COMPLETE              11.82
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2_20240  COMPLETE               8.87
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v17_p8_plot  COMPLETE              24.91
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_HRRR_suite_HRRR_2024091  COMPLETE              42.96
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0_20240916200  COMPLETE              26.50
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_GFS_v16_2024091620042  COMPLETE              44.27
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE             159.33
```

- [ ] derecho.intel
- [ ] gaea.intel
- [X] hera.gnu
- [ ] hera.intel
- [ ] hercules.intel
- [ ] jet.intel
- [ ] orion.intel
- [ ] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [ ] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)

## DEPENDENCIES:
N/A

## DOCUMENTATION:
All documentation

## ISSUE: 
Resolves Issue #1130 
Resolves Issue #340 

## CHECKLIST
- [X] My code follows the style guidelines in the Contributor's Guide
- [x] I have performed a self-review of my own code using the Code Reviewer's Guide
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain).
- [X] My changes generate no new warnings
- [X] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published N/A

### CODE REVIEW

File: ush/create_aqm_rc_file.py [hunk 1]
```diff
@@ -19,74 +19,81 @@
...
+
 def create_aqm_rc_file(cdate, run_dir, init_concentrations):
-    """ Creates an aqm.rc file in the specified run directory
+    """Creates an ``aqm.rc`` file in the specified run directory
 
     Args:
-        cdate: cycle date
-        run_dir: run directory
-        init_concentrations
+        cdate                (str): cycle date
```
Comment (MichaelLueken) [inline ush/create_aqm_rc_file.py]: Should the c in cycle be capitalized?

```suggestion
        cdate                (str): Cycle date
```

File: ush/create_aqm_rc_file.py [hunk 2]
```diff
@@ -19,74 +19,81 @@
...
+        run_dir              (str): run directory
```
Comment (MichaelLueken) [inline ush/create_aqm_rc_file.py]: Should the r in run be capitalized?

```suggestion
        run_dir              (str): Run directory
```

File: ush/create_diag_table_file.py [hunk 3]
```diff
@@ -21,12 +20,12 @@
 
 
 def create_diag_table_file(run_dir):
-    """Creates a diagnostic table file for each cycle to be run
+    """Creates an FV3 diagnostic table (``diag_table``) file for each cycle to be run
 
     Args:
-        run_dir: run directory
+        run_dir (str): run directory
```
Comment (MichaelLueken) [inline ush/create_diag_table_file.py]: Should the r in run be capitalized?

```suggestion
        run_dir (str): Run directory
```

File: ush/python_utils/environment.py [hunk 4]
```diff
@@ -8,12 +8,12 @@
 
 
 def str_to_date(s):
-    """Get python datetime object from string.
+    """Gets Python datetime object from string.
 
     Args:
-        s: a string
+        s (str): a string
```
Comment (MichaelLueken) [inline ush/python_utils/environment.py]: Should the a be capitalized?

```suggestion
        s (str): A string
```

File: ush/python_utils/environment.py [hunk 5]
```diff
@@ -32,31 +32,30 @@ def str_to_date(s):
...
 def date_to_str(d, format="%Y%m%d%H%M"):
-    """Get string from python datetime object.
-    By default it converts to YYYYMMDDHHMM format unless
-    told otherwise by passing a different format
+    """Gets string from Python datetime object.
+    By default it converts to ``YYYYMMDDHHmm`` format unless told otherwise by passing a different format.
 
     Args:
-        d: datetime object
+        d (datetime.datetime): datetime object
```
Comment (MichaelLueken) [inline ush/python_utils/environment.py]: Should the d in datetime be capitalized?

```suggestion
        d (datetime.datetime): Datetime object
```

File: ush/python_utils/environment.py [hunk 6]
```diff
@@ -32,31 +32,30 @@ def str_to_date(s):
...
     This will be used to automatically convert environment variables to data types
     that are more convenient to work with. If you don't want this functionality,
-    pass return_string = 1
+    pass ``return_string = 1``.
 
     Args:
-        s: a string
-        return_string: Set to 1 to return the string itself
-                       Set to 2 to return the string itself only for a datetime object
+        s (str): a string
```
Comment (MichaelLueken) [inline ush/python_utils/environment.py]: Should the a be capitalized?

```suggestion
        s (str): A string
```

File: ush/python_utils/environment.py [hunk 7]
```diff
@@ -111,13 +109,14 @@ def type_to_str(v):
...
 
 def list_to_str(v, oneline=False):
-    """Given a string or list of string, construct a string
-    to be used on right hand side of shell environement variables
+    """Given a string or list of strings, constructs a string
+    to be used on right hand side of shell environment variables.
 
     Args:
-        v: a string/number, list of strings/numbers, or null string('')
+        v: a string/number, list of strings/numbers, or null string(``''``)
```
Comment (MichaelLueken) [inline ush/python_utils/environment.py]: Should the a be capitalized?

```suggestion
        v: A string/number, list of strings/numbers, or null string(``''``)
```

File: ush/python_utils/environment.py [hunk 8]
```diff
@@ -111,13 +109,14 @@ def type_to_str(v):
...
+        oneline (bool): If the string is a single line (True) or multiple (False) ?
     Returns:
-        A string
+        a string
```
Comment (MichaelLueken) [inline ush/python_utils/environment.py]: Should the a be capitalized?

```suggestion
        A string
```

File: ush/python_utils/filesys_cmds_vrfy.py [hunk 9]
```diff
@@ -5,13 +5,13 @@
...
+    """Executes system command
 
     Args:
-        cmd: the command
-        *args: its arguments
+        cmd (str): The command
+        *args: Its arguments
     Returns:
-        Exit code
+        ret - Exit code
```
Comment (MichaelLueken) [inline ush/python_utils/filesys_cmds_vrfy.py]: Should the hyphen be replaced with a colon?

```suggestion
        ret: Exit code
```

File: ush/python_utils/run_command.py [hunk 10]
```diff
@@ -4,10 +4,10 @@
 
 
 def run_command(cmd):
-    """Run system command in a subprocess
+    """Runs system command in a subprocess
 
     Args:
-        cmd: command to execute
+        cmd (str): command to execute
```
Comment (MichaelLueken) [inline ush/python_utils/run_command.py]: Should the c in command be capitalized?

```suggestion
        cmd (str): Command to execute
```

File: ush/retrieve_data.py [hunk 11]
```diff
@@ -918,9 +1001,15 @@ def get_ens_groups(members):
 def parse_args(argv):
 
     """
-    Function maintains the arguments accepted by this script. Please see
-    Python's argparse documenation for more information about settings of each
-    argument.
+    Maintains the arguments accepted by this script. Please see Python's 
+    `argparse <https://docs.python.org/3/library/argparse.html>`_ documenation for more
```
Comment (MichaelLueken) [inline ush/retrieve_data.py]: documentation

```suggestion
    `argparse <https://docs.python.org/3/library/argparse.html>`_ documentation for more 
```

File: ush/run_srw_tests.py [hunk 12]
```diff
@@ -5,17 +5,19 @@
...
-# After the initial launch, the checkTests method is called to monitor the
-# status of each test and call the launch_FV3LAM_wflow.sh script repeatedly 
-# in each uncompleted workflow until all workflows are done.
 class SRWTest:
+
+  """Python class to handle the launching of a set of SRW tests.
+  The expectation is to have a "clean" experiment directory with only new experiments
+  that are ready to run (e.g., no ``_old*`` experiments left around from previous tests).
+  This script takes only one parameter (``-e`` or ``--exptdir``) which points to the 
+  ``expt_basedir`` specified when the ``run_WE2E_tests.sh`` is run to set up the tests.
```
Comment (MichaelLueken) [inline ush/run_srw_tests.py]: run_WE2E_tests.py

```suggestion
  ``expt_basedir`` specified when the ``run_WE2E_tests.py`` script is run to set up the tests.
```

**Review by MichaelLueken — CHANGES_REQUESTED** [review]
@gspetro-NOAA -

Thanks for adding a preliminary update to add technical documentation to the majority of the `ush` and `tests/WE2E` python scripts!  Some of the input arguments and return definitions are lower case instead of capitalized, so I noted the ones that I saw during my initial review.  A misspelt `documenation` was also noted during my review, as well as the modification of `run_WE2E_tests.sh` to `run_WE2E_tests.py` (the old shell script was removed after `run_WE2E_tests.py` was complete).

**Review by MichaelLueken — APPROVED** [review]
@gspetro-NOAA -

Thank you very much for making my requested changes!  This work looks good to me.

The AQM and fundamental tests were run and successfully passed:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
aqm_grid_AQM_NA13km_suite_GFS_v16_20240918090657                   COMPLETE            7843.59
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE            7843.59
```
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta_2  COMPLETE              11.81
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2_20240  COMPLETE               8.91
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v17_p8_plot  COMPLETE              25.19
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_HRRR_suite_HRRR_2024091  COMPLETE              42.46
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0_20240918162  COMPLETE              26.63
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_GFS_v16_2024091816270  COMPLETE              45.41
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE             160.41
```
Approving PR now.

**Comment by EdwardSnyder-NOAA** [comment]
Changes look good to me.
```
Calculating core-hour usage and printing final summary
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used
----------------------------------------------------------------------------------------------------
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta_2  COMPLETE              26.71
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2_20240  COMPLETE              19.77
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v17_p8_plot  COMPLETE              50.03
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_HRRR_suite_HRRR_2024092  COMPLETE              55.93
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0_20240926073  COMPLETE              41.41
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_GFS_v16_2024092607313  COMPLETE              91.24
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE             285.09

Detailed summary written to /glade/work/esnyder/pr-1131/expt_dirs/WE2E_summary_20240927090333.txt
```

**Review by EdwardSnyder-NOAA — APPROVED** [review]

**Comment by MichaelLueken** [comment]
The Jenkins tests successfully passed on Gaea, Hercules, and Orion.  The Derecho pipeline was down and the tests on Hera GNU, Hera Intel, and Jet were aborted after running over 8 hours.

I'm in the process of running the coverage WE2E tests on these machines now.

**Comment by MichaelLueken** [comment]
Moving into October, the NRAL0032 account is no longer useable.  Will run the Derecho coverage tests on Hercules.  Hera is down for maintenance.  Will update PR with Hera Intel coverage test results.  Jet WE2E coverage tests are still running this morning.

**Comment by MichaelLueken** [comment]
The Derecho coverage WE2E tests were run on Hercules and all tests successfully passed:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
custom_ESGgrid_IndianOcean_6km_20241001075245                      COMPLETE              29.30
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_plot_20  COMPLETE              72.38
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_GFS_v16_2024100107524  COMPLETE              52.26
grid_RRFS_CONUScompact_13km_ics_HRRR_lbcs_RAP_suite_HRRR_20241001  COMPLETE              55.30
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta_2  COMPLETE              14.08
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_HRRR_suite_HRRR_2024100107524  COMPLETE              26.15
pregen_grid_orog_sfc_climo_20241001075250                          COMPLETE              11.52
specify_template_filenames_20241001075252                          COMPLETE              12.39
2019_hurricane_barry_20241001075252                                COMPLETE              46.69
2019_memorial_day_heat_wave_20241001075253                         COMPLETE              46.03
2020_denver_radiation_inversion_20241001075254                     COMPLETE              47.32
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE             413.42
```

**Comment by MichaelLueken** [comment]
Both the fundamental and coverage WE2E tests are just sitting in queue on Jet, so I decided to run the Jet WE2E coverage tests on Hera.  All tests successfully passed:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used
----------------------------------------------------------------------------------------------------
community_20241002133730                                           COMPLETE              22.63
custom_ESGgrid_20241002133731                                      COMPLETE              25.66
custom_ESGgrid_Great_Lakes_snow_8km_20241002133732                 COMPLETE              20.51
custom_GFDLgrid_20241002133733                                     COMPLETE              11.42
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_nemsio_2021032018_202410  COMPLETE              11.40
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_netcdf_2022060112_48h_20  COMPLETE              83.08
get_from_HPSS_ics_RAP_lbcs_RAP_20241002133735                      COMPLETE              20.04
grid_RRFS_AK_3km_ics_FV3GFS_lbcs_FV3GFS_suite_HRRR_20241002133736  COMPLETE             475.19
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_plot_20  COMPLETE              60.26
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2_20241  COMPLETE               7.11
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_RRFS_v1beta_2024  COMPLETE             731.84
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE            1469.14
```
The Hera Intel WE2E coverage tests have successfully passed:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used
----------------------------------------------------------------------------------------------------
custom_ESGgrid_Peru_12km_20240930200120                            COMPLETE              36.51
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_grib2_2019061200_2024093  COMPLETE               9.35
get_from_HPSS_ics_GDAS_lbcs_GDAS_fmt_netcdf_2022040400_ensemble_2  COMPLETE            1532.58
get_from_HPSS_ics_HRRR_lbcs_RAP_20240930200122                     COMPLETE              17.89
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2_20240  COMPLETE               7.45
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_plot_20  COMPLETE              24.31
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_RAP_suite_RAP_20240930200125  COMPLETE              13.72
grid_RRFS_CONUS_25km_ics_GSMGFS_lbcs_GSMGFS_suite_GFS_v15p2_20240  COMPLETE               9.59
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2_202409  COMPLETE             456.01
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_20240930  COMPLETE             606.59
grid_RRFS_CONUScompact_3km_ics_HRRR_lbcs_RAP_suite_HRRR_202409302  COMPLETE             906.17
pregen_grid_orog_sfc_climo_20240930200129                          COMPLETE              10.08
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE            3630.25
```
as well as the Hera GNU WE2E coverage tests:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used
----------------------------------------------------------------------------------------------------
custom_ESGgrid_Central_Asia_3km_20241002122754                     COMPLETE              71.24
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_nemsio_2019061200_202410  COMPLETE              14.17
get_from_NOMADS_ics_FV3GFS_lbcs_FV3GFS_20241002122756              COMPLETE              20.32
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_HRRR_2024100212  COMPLETE              76.71
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_RRFS_v1beta_202  COMPLETE              30.66
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0_20241002122  COMPLETE              37.37
long_fcst_20241002122759                                           COMPLETE             136.52
MET_verification_only_vx_20241002122800                            COMPLETE               0.39
MET_ensemble_verification_only_vx_time_lag_20241002122802          COMPLETE               9.90
2019_halloween_storm_20241002122804                                COMPLETE              87.04
2020_jan_cold_blast_20241002122805                                 COMPLETE              85.39
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE             569.71
```
Moving forward with merging this work now.

**Pull request merged**